### PR TITLE
remove p=(+/-)math.inf case for paddle.nn.functional.pairwise_distance()

### PR DESCRIPTION
**Pull request by wangyuwen1999**
our pairwise_distance's backward implementation is different from the implementation of torch.
torch:
3a878158a4feda1a7023c88b0899f0db
paddle:
47c63fbf8567c07e2523d896b6edabc3
the biggest difference is torch will divide grad by mask.sum(dim, true), but paddle don't.

### CODE REVIEW

**Review by cangtianhuang — APPROVED** [review]
LGTM
